src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -0,0 +1,483 @@
+
+                var indexOfLastDot = fullyQualifiedContainerName.LastIndexOf('.');
+                var qualifiedNamespaceName = indexOfLastDot > 0 ? fullyQualifiedContainerName.Substring(0, indexOfLastDot) : string.Empty;
+
+                if (namespaceFilter.Contains(qualifiedNamespaceName))
+                {
+                    continue;
+                }
+
+                // Container of extension method (static class in C# and Module in VB) can't be generic or nested.
+                // Note that we might incorrectly ignore valid types, because, for example, calling `GetTypeByMetadataName` 
+                // would return null if we have multiple definitions of a type even though only one is accessible from here
+                // (e.g. an internal type declared inside a shared document).
+                var containerSymbol = compilation.GetTypeByMetadataName(fullyQualifiedContainerName);
+
+                if (containerSymbol != null)
+                {
+                    GetItemsFromTypeContainsPotentialMatches(containerSymbol, qualifiedNamespaceName, methodNames, receiverTypeSymbol, semanticModel, position, counter, builder);
+                }
+                else
+                {
+                    conflictTypeRootNode.Add(fullyQualifiedContainerName, (qualifiedNamespaceName, methodNames));
+                }
+            }
+
+            var ticks = Environment.TickCount;
+
+            GetItemsFromConflictingTypes(rootNamespaceSymbol, conflictTypeRootNode, builder, receiverTypeSymbol, semanticModel, position, counter);
+
+            counter.GetSymbolExtraTicks = Environment.TickCount - ticks;
+
+            return builder.ToImmutable();
+        }
+
+        private static void GetItemsFromTypeContainsPotentialMatches(
+            INamedTypeSymbol containerSymbol,
+            string qualifiedNamespaceName,
+            MultiDictionary<string, string>.ValueSet methodNames,
+            ITypeSymbol receiverTypeSymbol,
+            SemanticModel semanticModel,
+            int position,
+            StatisticCounter counter,
+            ArrayBuilder<SerializableImportCompletionItem> builder)
+        {
+            counter.TotalTypesChecked++;
+
+            if (containerSymbol == null ||
+                !containerSymbol.MightContainExtensionMethods ||
+                !IsSymbolAccessible(containerSymbol, position, semanticModel))
+            {
+                return;
+            }
+
+            foreach (var methodName in methodNames)
+            {
+                var methodSymbols = containerSymbol.GetMembers(methodName).OfType<IMethodSymbol>();
+
+                foreach (var methodSymbol in methodSymbols)
+                {
+                    counter.TotalExtensionMethodsChecked++;
+                    IMethodSymbol? reducedMethodSymbol = null;
+
+                    if (methodSymbol.IsExtensionMethod &&
+                        (methodNames.Count == 0 || methodNames.Contains(methodSymbol.Name)) &&
```

Comment:
i don't understand teh `methodNames.Count == 0` check.  if we had zero method names, how would we get into the outer loop in the first place.  I also don't understand the inner check.  We only asked for methods with this speicfic name (from methodNames), so how would methodNames *not* contain the method name??

Reply:
Good catch. It was a leftover from previous logic. Removed!